packages/fluentui/react-northstar/src/components/Segment/Segment.tsx

```diff
@@ -23,51 +32,84 @@ export interface SegmentProps
+  accessibility?: Accessibility<never>;
 
   /** An segment can show it is currently unable to be interacted with. */
   disabled?: boolean;
 
   /** A segment can have its colors inverted for contrast. */
   inverted?: boolean;
+
 }
 
+export type SegmentStylesProps = Required<Pick<SegmentProps, 'color' | 'inverted' | 'disabled'>>;
+
 export const segmentClassName = 'ui-segment';
 
-class Segment extends UIComponent<WithAsProp<SegmentProps>, any> {
-  static deprecated_className = segmentClassName;
+const Segment: React.FC<WithAsProp<SegmentProps>> & FluentComponentStaticProps<SegmentProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Segment.displayName, context.telemetry);
+  setStart();
+  const { children, content, color, inverted, disabled, className, design, styles, variables } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Segment.handledProps, props);
 
-  static displayName = 'Segment';
+  const getA11yProps = useAccessibility<never>(props.accessibility, {
+    debugName: Segment.displayName,
+    rtl: context.rtl,
+  });
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: 'shorthand',
-      color: true,
+  const { classes } = useStyles<SegmentStylesProps>(Segment.displayName, {
+    className: segmentClassName,
+    mapPropsToStyles: () => ({
+      color,
+      inverted,
+      disabled,
     }),
-    disabled: PropTypes.bool,
-    inverted: PropTypes.bool,
-    rtlAttributes: PropTypes.func,
-  };
-
-  static defaultProps = {
-    as: 'div',
-  };
-
-  renderComponent({ accessibility, ElementType, classes, unhandledProps }) {
-    const { children, content } = this.props;
-
-    return (
-      <ElementType
-        {...rtlTextContainer.getAttributes({ forElements: [children] })}
-        {...accessibility.attributes.root}
-        {...unhandledProps}
-        className={classes.root}
-      >
-        {childrenExist(children) ? children : Box.create(content)}
-      </ElementType>
-    );
-  }
-}
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const element = getA11yProps.unstable_wrapWithFocusZone(
+    <ElementType
+      {...getA11yProps('root', {
+        className: classes.root,
+        ...rtlTextContainer.getAttributes({ forElements: [children] }),
+        ...unhandledProps,
+      })}
+    >
+      {childrenExist(children) ? children : Box.create(content)}
+    </ElementType>,
+  );
+
+  setEnd();
+  return element;
+};
+
+Segment.displayName = 'Segment';
+
+Segment.propTypes = {
+  ...commonPropTypes.createCommon({
+    content: 'shorthand',
+    color: true,
+  }),
+  disabled: PropTypes.bool,
+  inverted: PropTypes.bool,
+};
+Segment.handledProps = Object.keys(Segment.propTypes) as any;
+
+Segment.defaultProps = {
+  as: 'div',
```

Comment:
This is redundant, `as` defaults to `div`

Reply:
Removed